Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -890,33 +890,33 @@ namespace Babylon
             {
                 xr::Ray nativeRay{{0, 0, 0}, {0, 0, -1}};
                 auto originObject = m_origin.Value();
-                if (originObject.HasOwnProperty("x"))
+                if (originObject.Has("x"))
```

Comment:
I apologize that this isn't really part of this PR, but this series of checks seems pretty bizarre. I think these checks involve string operations, so this could potentially be surprisingly slow. Is it an expected case that we'll be trying to get rays from objects that don't have XYZ components? That seems like it should be an error.

Reply:
I can remove it.  Yes I added this for safety as on the Babylon.js side this object is user specified and can technically exclude these properties, though it does not conform to the XRRay object spec.